crates/trigger-http2/src/server.rs

```diff
@@ -333,6 +288,111 @@ impl HttpServer {
     }
 }
 
+/// Handles a routed HTTP trigger request.
+pub struct RequestHandler {
```

Comment:
Could you say more about this change? It isn't clear how it relates to the conformance tests.

Reply:
To be fair this was from an evolutionary dead end. I do think it helps make things a bit easier to follow (at the expense of being slightly more verbose) by only giving the `OutboundHttpInterceptor` what it needs and nothing more. It encapsulates just the component handling and leaves out all the other HTTP sever stuff to the encapsulating `HttpServer` types. 

I can revert it if you want me to though.

Reply:
Its mostly just really hard to review the diff :sweat_smile:

Reply:
Sorry! If you aren't able to review, I'll revert that change tomorrow.

Reply:
I've reverted that change so hopefully the diff is a bit easier to read. Unfortunately, there's one spot where git's diffing still gets a bit confused.

Reply:
Thanks!